timemachine/cpp/src/kernels/k_logsumexp.cu

```diff
@@ -21,4 +21,31 @@ void __global__ k_exp_sub_max(
 template void __global__ k_exp_sub_max<float>(const int, const float *, const float *, float *);
 template void __global__ k_exp_sub_max<double>(const int, const double *, const double *, double *);
 
+template <typename RealType>
+void __global__ k_segmented_exp_sub_max(
+    const int num_segments,
```

Comment:
It can be inferred as long as `num_segments == blockDim.y` however we are likely to eventually run into not being able to launch that many blocks (as we ran into with an early implementation of this). Don't think it is clear/safe to rely on the state of the block configuration for kernels, we have run into a bug because of this in https://github.com/proteneer/timemachine/pull/1168. 

This is why I have been adopting the while loop syntax for kernels. It allows us to be independent of the block configuration as well the kernels always processes all of the data rather than terminating early if the blocks were incorrectly configured.